### PR TITLE
feat: add eth_createBundleAccessList RPC method

### PR DESCRIPTION
**Pull request by 0xSt1ng3R**
added eth_createBundleAccessList RPC method - https://github.com/paradigmxyz/reth/issues/5748

### CODE REVIEW

File: crates/rpc/rpc-api/src/eth.rs [hunk 1]
```diff
@@ -162,6 +162,13 @@ pub trait EthApi {
         state_override: Option<StateOverride>,
     ) -> RpcResult<Vec<EthCallResponse>>;
 
+    #[method(name = "createBundleAccessList")]
+    async fn create_bundle_access_list(
```
Comment (mattsse) [inline crates/rpc/rpc-api/src/eth.rs:166]: can we get some docs here on what this actually does?

File: crates/rpc/rpc/src/eth/api/call.rs [hunk 2]
```diff
@@ -349,6 +349,63 @@ where
...
+                    let from = call.from.unwrap_or_default();
+                    let to = call.to.unwrap_or(from.create(db.basic_ref(from)?.unwrap_or_default().nonce));
+                    let initial = call.access_list.take().unwrap_or_default();
+
+                    let precompiles = get_precompiles(env.cfg.spec_id);
+                    let mut inspector = AccessListInspector::new(initial, from, to, precompiles);
+                    let (result, env) = inspect(&mut db, env, &mut inspector)?;
+
+                    result.result.ensure_success()?;
+
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/api/call.rs:396]: I think we need to commit the result to the state, right?

**Comment by 0xSt1ng3R** [comment]
looks ok?

**Comment by rkrasiuk** [comment]
@0xSt1ng3R two things:
1. we are open to having `eth_estimateGasBundle` method as it is included in the "canonical" implementation of the bundle API in `flashbots/builder`
2. having said that, access list inspection is not supported there and the reference you've linked in the issue is some fork of mev-geth. generally, we only support methods that are either on the roadmap or a part of existing canonical / widely used implementations

any extra methods you want on your node can be implemented through CLI & RPC extensions. see [an example](https://github.com/paradigmxyz/reth/blob/8e214602e50893e02fdacf3c0020ee7473dca507/examples/rpc-db/src/myrpc_ext.rs) for that
closing the PR

**Comment by 0xSt1ng3R** [comment]
@rkrasiuk 
Thank you for the comment; it seems logical. What is your proposal for supporting access list generation for bundles as a native implementation in reth?

**Pull request closed without merging**